pkg/controllers/subnet/subnet_controller.go

```diff
@@ -233,6 +273,43 @@ func (r *SubnetReconciler) updateSubnetStatus(obj *v1alpha1.Subnet) error {
 	return nil
 }
 
+// handleSharedSubnet handles a shared subnet with associated_resource label
+// It gets the NSX subnet based on the associated resource (first from cache, then from NSX API),
+// updates the status of the Subnet CR, and adds the subnet to the polling queue for regular status updates.
+func (r *SubnetReconciler) handleSharedSubnet(ctx context.Context, subnetCR *v1alpha1.Subnet, namespacedName client.ObjectKey, associatedResource string) (ctrl.Result, error) {
```

Comment:
in this func, handle the Subnet CR with updated NSX Subnet Info, 
when Subnet CR is deleted, we can remove it from PollingQueue